### PR TITLE
Fix: Support for lwip getip with espressif

### PR DESCRIPTION
**Pull request by alfred2g**
Support for lwip getip with espressif

Description
-----------
Add support for WIFI_GetIP with espressif over the LwIP stack

Checklist:
----------

- [ X ] I have tested my changes. No regression in existing tests.
- [ X ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by IP555** [comment]
New economy of  Family 8 starts and I really the head over this way and forever pay to me!

File: libraries/abstractions/wifi/test/iot_test_wifi.c [hunk 1]
```diff
@@ -895,6 +895,11 @@ TEST( Full_WiFi, AFQP_WiFiGetIP )
 
         /* Assert that the IP address is found. */
         TEST_ASSERT( *( ( uint32_t * ) ucIPAddr ) != 0 );
+
+        TEST_ASSERT( ( ucIPAddr[ 0 ] >= 0 ) && ( ucIPAddr[ 0 ] <= 255 ) );
```
Comment (gkwicker) [inline libraries/abstractions/wifi/test/iot_test_wifi.c:899]: `ucIPAddr` is defined as `uint8_t[]` on line 881, so lines 899-902 are [NOPs](https://en.wikipedia.org/wiki/NOP_(code)), please delete them.  `uint8_t` values are 0-255 by definition.

File: vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c [hunk 2]
```diff
@@ -1072,29 +1074,49 @@ WIFIReturnCode_t WIFI_Ping( uint8_t * pucIPAddr,
...
+#else /* running lwip */
+        tcpip_adapter_ip_info_t ipInfo;
+        int ret;
+
+        ret = tcpip_adapter_get_ip_info(TCPIP_ADAPTER_IF_STA, &ipInfo);
+        if (ret == ESP_OK)
+        {
+            xRetVal = eWiFiSuccess;
+            memcpy( pucIPAddr, &ipInfo.ip.addr, sizeof( ipInfo.ip.addr ) );
+            configPRINTF(("%s: local ip address is %d.%d.%d.%d\n",
```
Comment (gkwicker) [inline vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c:1099]: Why are we printing the IP address for lwip but not for FreeRTOS+TCP (after line 1089)?

File: vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c [hunk 3]
```diff
@@ -43,6 +44,7 @@
 #include "nvs_flash.h"
 #include "nvs.h"
 
+
```
Comment (aggarw13) [inline vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c:47]: Unnecessary line break?

**Pull request merged**